substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -330,32 +435,48 @@ private static long init(boolean nowrap) {
                 if (ret == ZLib.Z_MEM_ERROR()) {
                     throw new OutOfMemoryError();
                 } else {
-                    throw new InternalError();
+                    throw new InternalError(CTypeConversion.toJavaString(strm.msg()));
                 }
             }
         }
     }
 
-    // Inflater.c-Java_java_util_zip_Inflater_setDictionary(JNIEnv *env, jclass cls, jlong addr,
+    // (JDK 11)src/java.base/share/native/libzip/Inflater.c-static void doSetDictionary(JNIEnv *env, jlong addr, jbyte *buf, jint len)
+    private static void doSetDictionary(long addr, CCharPointer bytes, int len) {
```

Comment:
This method needs an `@Substitute` annotation.

Reply:
My bad: Believing the error message from the build.  Since the original of this method is a C file-static method, this method needs to move to being a static method in a `Util_java_util_zip_Inflater` class.

Similarly for `Deflater.doSetDictionary`.

Reply:
I don't see why, but OK.